src/client/Client.cc

```diff
@@ -8516,14 +8520,14 @@ int Client::readdir_r_cb(dir_result_t *d, add_dirent_cb_t cb, void *p,
 
   if (dirp->at_end())
     return 0;
-
+  int mask = caps | CEPH_STAT_RSTAT;
   if (dirp->offset == 0) {
     ldout(cct, 15) << " including ." << dendl;
     ceph_assert(diri->dentries.size() < 2); // can't have multiple hard-links to a dir
     uint64_t next_off = 1;
 
     int r;
-    r = _getattr(diri, caps, dirp->perms);
+    r = _getattr(diri, mask, dirp->perms);
```

Comment:
I think this makes it clear we don't need the mask variable here. Just do:
```
    r = getattr(diri, caps | CEPH_STAT_RSTAT, dirp->perms);
```

That's a bit more clear and "mask" is a generic name for a variable with this large a scope.

Reply:
It's fixed，thanks!